internal/bminventory/inventory.go

```diff
@@ -6741,11 +6744,19 @@ func (b *bareMetalInventory) V2DownloadInfraEnvFiles(ctx context.Context, params
 }
 
 func (b *bareMetalInventory) V2DownloadClusterCredentials(ctx context.Context, params installer.V2DownloadClusterCredentialsParams) middleware.Responder {
+	fileName := params.FileName
 	respBody, contentLength, err := b.V2DownloadClusterCredentialsInternal(ctx, params)
+
+	if err != nil && params.FileName == constants.Kubeconfig {
```

Comment:
Add a comment on why this is necessary.

Reply:
Consider moving this logic into `V2DownloadClusterCredentialsInternal`

Reply:
why not to verify if object exists?

Reply:
@tsorya @eranco74 @romfreiman The kubeconfig (with ca) isn't available before the finalize stage so we should downgrade to the no-ingress kubeconfig. on the other hand it changes the API.
It's just an option, still consulting with @avishayt

Reply:
We had a dedicated API for getting the kubeconfig file that did a similar logic but we want to keep it simple and have only single generic API for downloading files so we deprecated it.
The Kubeconfig ingress file is required only for oauth logic so for the most users the no-igress kubeconfig is enough.
The idea here is to provide the best kubeconfig we have, so at the beginning we provide the no-ingress kubeconfig and after the finalize stage we download the kubeconfig with the ingress ca.

Reply:
@eranco74 I prefer keep the internal method simple if possible and do this fallback on the external API layer. 
In addition please note the downloaded file name is set by the external method.
`	return filemiddleware.NewResponder(installer.NewV2DownloadClusterCredentialsOK().WithPayload(respBody), fileName, contentLength)`


(anyway, and external method sets the downloaded file name)

Reply:
@tsorya right, The V2DownloadClusterCredentialsInternal method do both: checks the file existence and downloads it.

Reply:
So please check if the error is not-found, because now you will fall back for every kind of error

Reply:
@avishayt now we have two potential errors StatusBadRequest or StatusConflict one for file not found and the second for wrong status and in the future we might have more errors to deal with. prefer keep the code simple and clean with no contract between the two methods. in the worse case we will check something twice.